graphql/resolvers.go

```diff
@@ -2801,6 +2819,118 @@ func (r *versionResolver) BuildVariants(ctx context.Context, v *restModel.APIVer
+	if tasks == nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	timeTaken, makespan := task.GetTimeSpent(tasks)
+
+	// return nil if rounded timeTaken/makespan == 0s
+	t := timeTaken.Round(time.Second)
+	m := makespan.Round(time.Second)
+	if t.String() == "0s" {
+		return nil, nil
+	}
+	apiTimeTaken := restModel.NewAPIDuration(t)
+	apiMakespan := restModel.NewAPIDuration(m)
+	return &VersionTiming{
+		TimeTaken: &apiTimeTaken,
+		Makespan:  &apiMakespan,
+	}, nil
+}
+
+func (r *versionResolver) BaseVersionID(ctx context.Context, obj *restModel.APIVersion) (*string, error) {
+	baseVersion, err := model.VersionFindOne(model.BaseVersionByProjectIdAndRevision(*obj.Project, *obj.Revision).Project(bson.M{model.VersionIdentifierKey: 1}))
+	if baseVersion == nil || err != nil {
+		return nil, nil
+	}
+	return &baseVersion.Id, nil
+}
+
+func (r *versionResolver) ProjectIdentifier(ctx context.Context, obj *restModel.APIVersion) (string, error) {
+	return utility.FromStringPtr(obj.ProjectIdentifier), nil
+}
+
+func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion) (string, error) {
+	failedAndAbortedStatuses := append(evergreen.TaskFailureStatuses, evergreen.TaskAborted)
+	opts := data.TaskFilterOptions{
+		Statuses:        failedAndAbortedStatuses,
+		FieldsToProject: []string{task.DisplayStatusKey},
+	}
+	tasks, _, err := r.sc.FindTasksByVersion(*obj.Id, opts)
+	if err != nil {
+		return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
+	}
+	status, _ := evergreen.VersionStatusToPatchStatus(*obj.Status)
+
+	if err := data.ValidatePatchID(*obj.Id); err == nil {
+		p, err := r.sc.FindPatchById(*obj.Id)
+		if err != nil {
+			return status, InternalServerError.Send(ctx, fmt.Sprintf("Something went wrong `%s`", err))
+		}
+
+		if len(p.ChildPatches) > 0 {
+			allPatches := []restModel.APIPatch{}
+			allPatches = append(allPatches, *p)
+			for _, cp := range p.ChildPatches {
+				allPatches = append(allPatches, cp)
+				// add the child patch tasks to tasks so that we can consider their status
+				childPatchTasks, _, err := r.sc.FindTasksByVersion(*cp.Id, opts)
+				if err != nil {
+					return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
+				}
+				tasks = append(tasks, childPatchTasks...)
+			}
+			// determine what the main patch status should be given the status of
+			// the child patches
+			sort.Sort(restModel.PatchesByStatus(allPatches))
+
+			// after sorting, the first patch will be the one with the highest priority
+			status = *allPatches[0].Status
+		}
+	}
+	statuses := getAllTaskStatuses(tasks)
+
+	// If theres an aborted task we should set the patch status to aborted if there are no other failures
+	if utility.StringSliceContains(statuses, evergreen.TaskAborted) {
+		if len(utility.StringSliceIntersection(statuses, evergreen.TaskFailureStatuses)) == 0 {
+			abortedStatus := evergreen.TaskAborted
```

Comment:
It feels like bad style to use a Task constant for a patch status. Can we consider making a PatchAborted?